src/t8_forest/t8_forest_cxx.cxx

```diff
@@ -1372,55 +1555,127 @@ t8_forest_populate (t8_forest_t forest)
 static int
 t8_forest_tree_shared (t8_forest_t forest, int first_or_last)
 {
+  T8_ASSERT (t8_forest_is_committed (forest));
+  T8_ASSERT (first_or_last == 0 || first_or_last == 1);
+  T8_ASSERT (forest != NULL);
+  T8_ASSERT (forest->first_local_tree > -1);
+  T8_ASSERT (forest->first_local_tree < forest->global_num_trees);  
+  T8_ASSERT ((forest->first_local_tree == 0 && forest->last_local_tree == -1) ?
```

Comment:
Please add documentation to non-trivial assertions.
Can you split this assertion into multiple assertions? We prefer readable code over super-compact code.